test/pop_protocol_test.erl

```diff
@@ -116,3 +140,30 @@ basic_test() ->
     ?assertError(_, lists:foldl(FoldFn, PD0, [B1, B1T])), 
 
     ok.
+
+%% get entry in block's consensus data
+get_block_cd(Entry, Block) -> maps:get(Entry, maps:get(consensus_data, Block)).
+
+
+generate_block_test() ->
+    %% ?debugHere,
+    %% ?debugVal("hi"),
+    {VerifierArr, PrivateKey, _} = make_verifier_array(),
+
+    CurrentTime = 100,
+
+    PD0 = pop_protocol:initialize_protocol_data(VerifierArr, 10, 5, hype_chain, CurrentTime),
+
+    MakeChainFn = fun(VerifierIndex, PD) -> 
+			  Block = generate_block(VerifierIndex, PrivateKey, PD),
+			  %% io:format(user, "B ~p ~n", [Block]),
+			  %% io:format(user, "~n~n", []),
+			  pop_protocol:add_new_block(Block, get_block_cd(timestamp, Block) + 1, PD) 
+		  end,
+
+    lists:foldl(MakeChainFn, PD0, [1]),
+    lists:foldl(MakeChainFn, PD0, [1, 0]),
+    lists:foldl(MakeChainFn, PD0, [1, 1]),
+    lists:foldl(MakeChainFn, PD0, [1, 0, 2, 4, 0, 3, 1]),
```

Comment:
cool!

Reply:
should this test do any sort of checking?

Reply:
i do some in the other function, this is just to check that you can chain this thing well

Reply:
or something